core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -321,7 +321,7 @@ private[spark] class Executor(
           logInfo(s"Executor killed $taskName (TID $taskId)")
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled))
 
-        case cDE: CommitDeniedException =>
+        case CausedBy(cDE: CommitDeniedException) =>
```

Comment:
It's tidy, but seems like overkill if this is the only place in the code that checks if "T, or its immediate cause, is a FooException". `case t: Throwable if t.getCause.isInstanceOf[CommitDeniedException] =>` is sufficient to handle the additional case, at the cost of repeating the line of code.

Alternatively, if there are really a few instances of this pattern in the code that we can clean up with this pattern, then it seems worthwhile.